### PR TITLE
feat(openapi): use OpenAPI Generator instead of NSwagStudio

### PR DESCRIPTION
**Pull request by CBenoit**
This also add a workflow to publish the generated clients.

### CODE REVIEW

File: devolutions-gateway/openapi/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# REST API clients
+
+Clients to use the Devolutions Gateway REST API.
+
+C# and TypeScript clients are generated using [OpenAPI Generator](https://openapi-generator.tech/) and OpenAPI documents.
+
+## Generate clients code
+
+1. Make OpenAPI documents are up to date by running `TODO`
```
Comment (thenextman) [inline devolutions-gateway/openapi/README.md:9]: nit "Make" > "Make sure that"

Comment (CBenoit) [reply devolutions-gateway/openapi/README.md:9]: Oh, forgot about the TODO also, thanks!

File: devolutions-gateway/src/openapi.rs [hunk 2]
```diff
@@ -136,3 +136,22 @@ impl Modify for SubscriberSecurityAddon {
...
+/// Process a message originating from a Devolutions Gateway instance
+#[allow(unused)]
+#[utoipa::path(
+    post,
+    operation_id = "PostMessage",
+    tag = "Subscriber",
+    path = "/dgw/subscriber",
+    request_body(content = SubscriberMessage, description = "Message", content_type = "application/json"),
+    responses(
+        (status = 200, description = "Message received and processed successfuly"),
```
Comment (thenextman) [inline devolutions-gateway/src/openapi.rs:149]: typo: "successfuly" > "successfully"

**Review by thenextman — APPROVED** [review]
LGTM

**Comment by CBenoit** [comment]
I addressed the comments, thank you!

**Pull request merged**